api/clients/v2/eigenda_client.go

```diff
@@ -0,0 +1,279 @@
+	clientConfig *EigenDAClientConfig
+	codec        codecs.BlobCodec
+	relayClient  RelayClient
+	g1Srs        []bn254.G1Affine
+	blobVerifier verification.IBlobVerifier
+}
+
+// BuildEigenDAClient builds an EigenDAClient from config structs.
+func BuildEigenDAClient(
+	log logging.Logger,
+	clientConfig *EigenDAClientConfig,
+	ethConfig geth.EthClientConfig,
+	relayClientConfig *RelayClientConfig,
+	g1Srs []bn254.G1Affine) (*EigenDAClient, error) {
+
+	relayClient, err := NewRelayClient(relayClientConfig, log)
+	if err != nil {
+		return nil, fmt.Errorf("new relay client: %w", err)
+	}
+
+	ethClient, err := geth.NewClient(ethConfig, gethcommon.Address{}, 0, log)
+	if err != nil {
+		return nil, fmt.Errorf("new eth client: %w", err)
+	}
+
+	blobVerifier, err := verification.NewBlobVerifier(ethClient, clientConfig.EigenDABlobVerifierAddr)
+	if err != nil {
+		return nil, fmt.Errorf("new blob verifier: %w", err)
+	}
+
+	codec, err := createCodec(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewEigenDAClient(
+		log,
+		rand.New(rand.NewSource(rand.Int63())),
+		clientConfig,
+		relayClient,
+		blobVerifier,
+		codec,
+		g1Srs)
+}
+
+// NewEigenDAClient assembles an EigenDAClient from subcomponents that have already been constructed and initialized.
+func NewEigenDAClient(
+	log logging.Logger,
+	random *rand.Rand,
+	clientConfig *EigenDAClientConfig,
+	relayClient RelayClient,
+	blobVerifier verification.IBlobVerifier,
+	codec codecs.BlobCodec,
+	g1Srs []bn254.G1Affine) (*EigenDAClient, error) {
+
+	return &EigenDAClient{
+		log:          log,
+		random:       random,
+		clientConfig: clientConfig,
+		codec:        codec,
+		relayClient:  relayClient,
+		blobVerifier: blobVerifier,
+		g1Srs:        g1Srs,
+	}, nil
+}
+
+// GetPayload iteratively attempts to fetch a given blob with key blobKey from relays that have it, as claimed by the
+// blob certificate. The relays are attempted in random order.
+//
+// If the blob is successfully retrieved, then the blob is verified. If the verification succeeds, the blob is decoded
+// to yield the payload (the original user data), and the payload is returned.
+func (c *EigenDAClient) GetPayload(
+	ctx context.Context,
+	blobKey core.BlobKey,
+	eigenDACert *verification.EigenDACert) ([]byte, error) {
+
+	err := c.verifyCertWithTimeout(ctx, eigenDACert)
+	if err != nil {
+		return nil, fmt.Errorf("verify cert with timeout for blobKey %v: %w", blobKey, err)
+	}
+
+	relayKeys := eigenDACert.BlobVerificationProof.BlobCertificate.RelayKeys
+	relayKeyCount := len(relayKeys)
+	if relayKeyCount == 0 {
+		return nil, errors.New("relay key count is zero")
+	}
+
+	blobCommitmentProto := contractEigenDABlobVerifier.BlobCommitmentBindingToProto(
+		&eigenDACert.BlobVerificationProof.BlobCertificate.BlobHeader.Commitment)
+	blobCommitment, err := encoding.BlobCommitmentsFromProtobuf(blobCommitmentProto)
+
+	if err != nil {
+		return nil, fmt.Errorf("blob commitments from protobuf: %w", err)
+	}
```

Comment:
feel like this should be a single util function that does both steps.
Also can we rename `blobCommitmentProto` to something like `blobCommitmentInCert`

Reply:
I added a utility function, and sidestepped the name `blobCommitmentProto` [7b66df6a](https://github.com/Layr-Labs/eigenda/pull/972/commits/7b66df6aa7168cfca6c6a3a5aab55c79370ef08b)

LMK what you think about the chosen name for the util function, it's awkward but nothing better came to mind

Reply:
LGTM, thanks! I don't like though that we use `commitmentS` (with an s) in one place and not the other.
![image](https://github.com/user-attachments/assets/5271eee3-f2a2-4a39-a262-c2338d926f34)

Can we make those the same (prob remove the s?)